app/static/src/app/store/fitData/actions.ts

```diff
@@ -0,0 +1,49 @@
+import { parse } from "csv-parse";
+import { ActionTree } from "vuex";
+import { AppState } from "../appState/state";
+import { FitDataState } from "./state";
+import { Error } from "../../types/responseTypes";
+import { FitDataMutation } from "./mutations";
+import { processFitData, ProcessFitDataResult } from "../../utils";
+import userMessages from "../../userMessages";
+
+export enum FitDataAction {
+    Upload = "Upload"
+}
+
+export const actions: ActionTree<FitDataState, AppState> = {
```

Comment:
There is no interaction with an endpoint here, could this not be in utils function or getters ?

Reply:
It feels like this is doing something equivalent to an api call - it's just getting data from a file upload rather than an endpoint, and the result of that is that new data can be committed via a mutation. So to me this feels like a pretty textboox action taken by the user from a component resulting in a change to the state. If you like, I could move the actual file upload stuff into a separate util but I think the whole thing should be managed from an action.

Reply:
I think upload could be placed in a more generic folder, separated from persistence. Since upload method could be re-used, I feel it's more organic to have it in utils. You could then handle persistence however.

Reply:
OK, I'll add a fileUpload util module and call that from the action